src/util/logger/formatter.rs

```diff
@@ -0,0 +1,251 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Copyright 2018 The Serde Json Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io;
+
+#[inline]
+fn is_escape_key(byte: u8) -> bool {
+    byte <= 0x20 || byte == 0x22 || byte == 0x3D || byte == 0x5B || byte == 0x5D
+}
+
+#[inline]
+fn need_escape(bytes: &[u8]) -> bool {
+    for &byte in bytes {
+        if is_escape_key(byte) {
+            return true;
+        }
+    }
+    false
+}
+
+pub fn write_escaped_str<W: ?Sized>(writer: &mut W, value: &str) -> io::Result<()>
```

Comment:
Why re-implement / copy a JSON implementation?

Reply:
At first, I try to use `serde_json::to_writer()` which writes and encodes values to a writer buf, but it will move the ownership of the writer buf. So, to keep the owernship, I have to copy thoes code😅

Reply:
How about just `serde_json::to_writer(writer, ...)`? AFAIK, `Write` is [implemented](https://doc.rust-lang.org/std/primitive.reference.html#impl-Write) for `&mut W: Write`, so your `&mut W` is already a `Writer`?